pkg/clientwrappers/dataflow/dataflow.go

```diff
@@ -84,21 +85,30 @@ func GetDataflowInformation(flow *v1.NifiDataflow, config *clientconfig.NifiConf
 
 // CreateDataflow will deploy the NifiDataflow on NiFi Cluster.
 func CreateDataflow(flow *v1.NifiDataflow, config *clientconfig.NifiConfig,
-	registry *v1.NifiRegistryClient) (*v1.NifiDataflowStatus, error) {
+	registry *v1.NifiRegistryClient, resource *v1alpha1.NifiResource) (*v1.NifiDataflowStatus, error) {
 	log.Debug("Creating dataflow",
 		zap.String("clusterName", flow.Spec.ClusterRef.Name),
 		zap.String("flowId", flow.Spec.FlowId),
 		zap.String("dataflow", flow.Name))
 
+	var parentProcessGroup *v1.ParentProcessGroup
+
+	if resource != nil && resource.Status.UUID != "" && resource.Spec.Type != "" {
+		parentProcessGroup = &v1.ParentProcessGroup{
+			UUID: resource.Status.UUID,
+			Type: resource.Spec.Type,
+		}
+	}
+
```

Comment:
```suggestion
	var parentProcessGroupId := ""

	if resource != nil && resource.Status.UUID != "" && resource.Spec.Type != "" {
		parentProcessGroupId = resource.Status.UUID
	}

```

Not sure `&& resource.Spec.Type != ""` is needed as you check before if it's a process group if it's not you should not do anything. So either you remove it or you need to make sure it's a process group I guess.